### ISSUE TITLE
Make it easier to amplify the sitemap.xml crawl

### ISSUE THREAD

**Issue by mgifford** [GovTechSG/purple-a11y#256]
I know that scanning a handful of sites is a problem. Scanning all web pages in a site also has problems associated with it. To have confidence in accessibility, we should be able to take a random sampling of pages, and be able to have some statistical certainty of our knowledge about the accessibility of a whole site.

The trouble is that there are often too many URLs, and you have site-maps of sitemaps.

To fine the list of URLs we can either crawl a site, or simply look for the Sitemap.xml file and based on that create a sub-set which will allow us to get a picture of a site’s accessibility. Now sitemap files can be gamed, and often they are used for more than SEO.

So I created a script which aggregates the XML files into just one and then removes all the stuff that we don’t want to be analyzing (.doc, .pdf, .zip, ext.) Now I have a random sampling of URLs. This XML file can be used in the future to test if the exact same URLs have improved over time (or not). I’m capping sitemap size at 2000 URLs as that is a pretty decent sampling for the sites we work with.

**Comment by mgifford** [comment]
Probably easiest to just see it here:   
https://github.com/mgifford/purple-a11y/blob/master/sitemap-tools/sitemap-randomizer.py
